Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp

```diff
@@ -410,7 +411,21 @@ namespace Multiplayer
                 editorServerLevelDataPacket.SetLastUpdate(true);
             }
 
-            connection->SendReliablePacket(editorServerLevelDataPacket);
+            // Try to send the packet to the Editor server. Retry if necessary.
+            bool packetSent = false;
+            static constexpr int MaxRetries = 20;
+            static constexpr int MillisecondDelayPerRetry = 100;
+            int numRetries = 0;
+            while (!packetSent && (numRetries < MaxRetries))
+            {
+                packetSent = connection->SendReliablePacket(editorServerLevelDataPacket);
+                if (!packetSent)
+                {
+                    AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(MillisecondDelayPerRetry));
```

Comment:
This would be a good spot to add `AZ::Interface<AzNetworking::INetworking>::Get()->ForceUpdate();`. You could check that the send buffer has enough space and go from there. Just be sure to pad that check since SendReliablePacket will have to add extra for the packet header.

Reply:
That's a great idea as a short-term solution, but I still think it needs to get addressed more holistically and at a lower level. Otherwise, everyone who calls SendReliablePacket will need to write the same error-checking and error-handling code on the off chance that their call is the one that causes the buffer to overflow. Perhaps we need an API like "ReliablySendReliablePacket". :P

Reply:
Side note - there's no obvious way to check the send buffer size without exposing more implementation details into the interface, but it gets caught indirectly anyways through the failure in SendReliablePacket.

Reply:
Agreed, it's worth a longer term rethink.